gcovr/gcov.py

```diff
@@ -25,6 +25,44 @@
 c_style_comment_pattern = re.compile('/\*.*?\*/')
 cpp_style_comment_pattern = re.compile('//.*?$')
 
+gcov_required_options = [
+    "--branch-counts",
+    "--branch-probabilities",
+    "--preserve-paths",
+    '--object-directory']
+
+
+class GCovSupportChecker(object):
+    """Check that gcov supports the options we need"""
+    def __init__(self):
+        self.checked = {}
+
+    def check(self, gcov_path):
+        if gcov_path not in self.checked:
+            cmd = gcov_path.split()
+            cmd.append("--help")
+            self.checked[gcov_path] = False
+            try:
+                help = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+            except subprocess.CalledProcessError as cpe:
+                help = cpe.output
+            help = str(help)
+            for option in gcov_required_options:
+                if option not in help:
+                    try:
+                        gcov_path = subprocess.check_output(
+                            ["which", gcov_path]).strip().decode("utf-8")
```

Comment:
Is the `which` utility available on Windows, or should the executable path only be resolved on non-Windows systems? I agree that resolving the path leads to better error messages when possible.

Reply:
on windows `type` does the same thing as `which` does. So that could be used instead. (Infact, `type` on linux does a very similar job too)

I did also notice that in the other uses of the `options.gcov_cmd` that if for some reason a user supplied a script with escaped or quoted whitespace then things would go wrong too.

Reply:
OK, then we can select between `which` and `type` depending on the OS. That should also fix the tests on Appveyor.

The `type` utility on Posixish systems is a shell builtin rather than a separate program, and cannot be used reliably to resolve the path of an executable (though `type -P $name` happens to do that in Bash).